Solutions/Salesforce Service Cloud/Workbooks/SalesforceServiceCloud.json

```diff
@@ -0,0 +1,659 @@
+                  },
+                  {
+                    "durationMs": 86400000
+                  },
+                  {
+                    "durationMs": 172800000
+                  },
+                  {
+                    "durationMs": 259200000
+                  },
+                  {
+                    "durationMs": 604800000
+                  },
+                  {
+                    "durationMs": 1209600000
+                  },
+                  {
+                    "durationMs": 2419200000
+                  },
+                  {
+                    "durationMs": 2592000000
+                  },
+                  {
+                    "durationMs": 5184000000
+                  },
+                  {
+                    "durationMs": 7776000000
+                  }
+                ],
+                "allowCustom": true
+              },
+              "timeContext": {
+                "durationMs": 86400000
+              }
+            }
+          ],
+          "style": "pills",
+          "queryType": 0,
+          "resourceType": "microsoft.operationalinsights/workspaces"
+        },
+        "name": "parameters - 32"
+      },
+      {
+        "type": 11,
+        "content": {
+          "version": "LinkItem/1.0",
+          "style": "tabs",
+          "links": [
+            {
+              "id": "ae90d1dc-20da-4948-80da-127b210bf152",
+              "cellValue": "view_tab",
+              "linkTarget": "parameter",
+              "linkLabel": "User Logins",
+              "subTarget": "1",
+              "style": "link"
+            },
+            {
+              "id": "af58b4d9-a888-43ed-91a9-6e9f539a61d4",
+              "cellValue": "view_tab",
+              "linkTarget": "parameter",
+              "linkLabel": "API Usage",
+              "subTarget": "2",
+              "style": "link"
+            }
+          ]
+        },
+        "name": "links - 34"
+      },
+      {
+        "type": 12,
+        "content": {
+          "version": "NotebookGroup/1.0",
+          "groupType": "editable",
+          "title": "User login locations",
+          "items": [
+            {
+              "type": 3,
+              "content": {
+                "version": "KqlItem/1.0",
+                "query": "let Countrydb = externaldata(Network:string, geoname_id:string, continent_code:string, continent_name:string, country_iso_code:string, country_name:string)\n[@\"https://raw.githubusercontent.com/datasets/geoip2-ipv4/master/data/geoip2-ipv4.csv\"];\nlet UsersLocation = SalesforceServiceCloud\n| where EventType == \"Login\"\n| project TimeGenerated, SourceIp;\nUsersLocation\n| extend Dummy=1\n| summarize count() by Hour=bin(TimeGenerated,24h), SourceIp,Dummy\n| partition by Hour(\n                  lookup (Countrydb|extend Dummy=1) on Dummy\n                | where ipv4_is_match(SourceIp, Network)\n                )\n| summarize sum(count_) by country_name",
```

Comment:
Do we have the permission to use this external source?

Reply:
yes